spring-integration-file/src/main/java/org/springframework/integration/file/aggregator/FileMarkerReleaseStrategy.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.file.aggregator;
+
+import java.util.Collection;
+
+import org.springframework.integration.aggregator.ReleaseStrategy;
+import org.springframework.integration.file.FileHeaders;
+import org.springframework.integration.file.splitter.FileSplitter;
+import org.springframework.integration.store.MessageGroup;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHeaders;
+
+/**
+ * A {@link ReleaseStrategy} which makes a decision based on the presence of
+ * {@link org.springframework.integration.file.splitter.FileSplitter.FileMarker.Mark#END}
+ * message in the group and its {@link org.springframework.integration.file.FileHeaders#LINE_COUNT} header.
+ *
+ * @author Artem Bilan
+ *
+ * @since 5.5
+ */
+public class FileMarkerReleaseStrategy implements ReleaseStrategy {
+
+	@Override
+	public boolean canRelease(MessageGroup group) {
+		int size = group.size();
+		if (size > 1) { // Need more than only a START marker
+			Collection<Message<?>> messages = group.getMessages();
+			for (Message<?> message : messages) {
```

Comment:
This is still no good; we can't iterate over the whole collection each time, it just won't scale with large groups.

Reply:
Correct, but see what we have so far in other place - `SequenceAwareMessageGroup`.
Not related to this one, but similar iteration on each added message.

Nevertheless I agree with your concern and really will address i, but I'd like to do that in the separate PR.
See my yesterday's comment.

Reply:
OK; but let's open an issue so it's not forgotten.

Reply:
If you agree with my proposal, I'll start working on that immediately, so this `FileAggregator` won't be a bottle neck any more.

Reply:
Yes; proposal looks good.